app/internal/security/ouath2.py

```diff
@@ -0,0 +1,124 @@
+from datetime import datetime, timedelta
+from typing import Union
+
+from app.config import JWT_ALGORITHM, JWT_KEY
+from passlib.context import CryptContext
+from app.database.database import SessionLocal
+from app.database.models import User
+from fastapi import Depends, HTTPException
+from fastapi.security import OAuth2PasswordBearer
+import jwt
+from starlette.requests import Request
+from starlette.responses import RedirectResponse
+from starlette.status import HTTP_401_UNAUTHORIZED
+from .schema import LoginUser
+
+
+JWT_MIN_EXP = 3
+pwd_context = CryptContext(schemes=["bcrypt"])
+oauth_schema = OAuth2PasswordBearer(tokenUrl="/login")
+
+
+async def get_db_user_by_username(username: str) -> User:
+    '''Checking database for user by username field'''
+    session = SessionLocal()
+    return session.query(User).filter_by(username=username).first()
+
+
+def get_hashed_password(password) -> str:
+    '''Hashing user password'''
+    return pwd_context.hash(password)
+
+
+def verify_password(plain_password, hashed_password) -> bool:
+    '''Verifying password and hashed password are equal'''
+    try:
+        return pwd_context.verify(plain_password, hashed_password)
+    except Exception:
+        return False
+
+
+async def authenticate_user(user: LoginUser) -> Union[LoginUser, bool]:
+    '''Verifying user is in database and password is correct'''
+    db_user = await get_db_user_by_username(username=user.username)
+    if db_user:
+        if verify_password(user.hashed_password, db_user.password):
+            return LoginUser(
+                username=user.username, hashed_password=db_user.password)
+    return False
+
+
+def create_jwt_token(user: LoginUser) -> str:
+    '''Creating jwt-token out of user unique data'''
+    expiration = datetime.utcnow() + timedelta(minutes=JWT_MIN_EXP)
+    jwt_payload = {
+        "sub": user.username,
+        "hashed_password": user.hashed_password,
+        "exp": expiration}
+    jwt_token = jwt.encode(
+        jwt_payload, JWT_KEY, algorithm=JWT_ALGORITHM)
+    return jwt_token
+
+
+async def check_jwt_token(
+    token: str = Depends(oauth_schema),
+        logged_in: bool = False, path: bool = None) -> Union[User, bool]:
+    '''
+    Check whether JWT token is correct. Returns User object if yes.
+    Returns None or raises HTTPException,
+    depanding which depandency activated this function.
+    '''
+    try:
+        jwt_payload = jwt.decode(
+            token, JWT_KEY, algorithms=JWT_ALGORITHM)
+        jwt_username = jwt_payload.get("sub")
+        jwt_hashed_password = jwt_payload.get("hashed_password")
+        db_user = await get_db_user_by_username(username=jwt_username)
```

Comment:
You usually don't want to talk with the database when checking the validity of the JWT token - that's one of the benefits of JWT.

Reply:
Validating the token is the lesser reason for that. I can do it by comparing to to the SECRET_KEY, and just by successfully decoding the token.
The main reason for that is: Already using the authentication function to return the User instance to the protected route. This way everybody can use it for other functionalities, without the need to query in each route. The dependency itself,  is already returning the User object with all its parameters.
Do you prefer that i will change that?